.hlint.yaml

```diff
@@ -0,0 +1,94 @@
+# https://github.com/ndmitchell/hlint
+##########################
+
+# This file contains a template configuration file, which is typically
+# placed as .hlint.yaml in the root of your project
+
+- arguments: [-XTypeApplications, -XQuasiQuotes, --cpp-define=CONFIG=dev]
+
+- ignore: {name: Redundant do}
+- ignore: {name: Redundant bracket}
+- ignore: {name: Redundant $}
+- ignore: {name: Redundant flip}
+- ignore: {name: Move brackets to avoid $}
+- ignore: {name: Eta reduce}
+- ignore: {name: Avoid lambda}
+- ignore: {name: Use camelCase}
+- ignore: {name: Use const}
+- ignore: {name: Use if}
+- ignore: {name: Use notElem}
+- ignore: {name: Use fromMaybe}
+- ignore: {name: Use maybe}
+- ignore: {name: Use fmap}
+- ignore: {name: Use foldl}
+- ignore: {name: 'Use :'}
+- ignore: {name: Use ++}
+- ignore: {name: Use ||}
+- ignore: {name: Use &&}
+- ignore: {name: 'Use ?~'}
+- ignore: {name: Use <$>}
+- ignore: {name: Use .}
+- ignore: {name: Use head}
+- ignore: {name: Use String}
+- ignore: {name: Use Foldable.forM_}
+- ignore: {name: Unused LANGUAGE pragma}
+- ignore: {name: Use newtype instead of data}
+# Rules not found in old HLint.hs file (prior to HLint 2.0)
+# Added when we made the change.
+- ignore: {name: Redundant lambda}
+- ignore: {name: Use section}
+
+# Specify additional command line arguments
+#
+# - arguments: [--color, --cpp-simple, -XQuasiQuotes]
+
+
+# Control which extensions/flags/modules/functions can be used
+#
+# - extensions:
+#   - default: false # all extension are banned by default
+#   - name: [PatternGuards, ViewPatterns] # only these listed extensions can be used
+#   - {name: CPP, within: CrossPlatform} # CPP can only be used in a given module
+#
+# - flags:
+#   - {name: -w, within: []} # -w is allowed nowhere
+#
+# - modules:
+#   - {name: [Data.Set, Data.HashSet], as: Set} # if you import Data.Set qualified, it must be as 'Set'
+#   - {name: Control.Arrow, within: []} # Certain modules are banned entirely
+#
+# - functions:
+#   - {name: unsafePerformIO, within: []} # unsafePerformIO can only appear in no modules
+
+
+# Add custom hints for this project
+#
+# Will suggest replacing "wibbleMany [myvar]" with "wibbleOne myvar"
+# - error: {lhs: "wibbleMany [x]", rhs: wibbleOne x}
+- warn: { lhs: "Cardano.Wallet.Util.traceBuildable", rhs: "()", note: "'traceBuildable' remains in code"}
+
+
+# Turn on hints that are off by default
+#
+# Ban "module X(module X) where", to require a real export list
+# - warn: {name: Use explicit module export list}
+#
+# Replace a $ b $ c with a . b $ c
+# - group: {name: dollar, enabled: true}
+#
+# Generalise map to fmap, ++ to <>
+# - group: {name: generalise, enabled: true}
+
+
+# Ignore some builtin hints
+# - ignore: {name: Use let}
+# - ignore: {name: Use const, within: SpecialModule} # Only within certain modules
+
+
+# Define some custom infix operators
+# - fixity: infixr 3 ~^#^~
+
+
+# To generate a suitable file for HLint do:
+# $ hlint --default > .hlint.yaml
```

Comment:
That's also a big topic of discussion actually :/ ... 
I'd suggest that for now, we do start with the _default_ hlint configuration, and, iterate over the weeks to add or remove stuff we like or don't.